sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    public partial class AnswerResult : IReadOnlyDictionary<string, object>
```

Comment:
Why is this change needed?

Reply:
Apologies for not writing a description in the PR before slapping it out of draft. Check now, and let me know if you have questions.

If Azure.Search.Documents has not gone GA yet, and you'd like to drop the IDictionary interface and just have AdditionalProperties public let me know and I'll revert that bit. We are assuming by default no API change unless asked.

Reply:
Thanks, got it. We can remove the `IDictionary` interface from `AnswerResult` and `CaptionResult` as they are not part of the [latest GA](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Search.Documents_11.3.0/sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs).

It should stay on `FacetResult`. However, shouldn't then there be no change in the one generated file? Or, if we are splitting it up, the new file can also be under Generated\Models\ (something like FacetResult.Dictionary.cs).

Reply:
Removed in those two as requested - https://github.com/Azure/azure-sdk-for-net/pull/21778/commits/2efb27f00e53716ca61091aaa9c23d2def789129

On FacetResult we add the elements no longer code gen'd [here](https://github.com/Azure/azure-sdk-for-net/blob/b8d45e2478f1404bc22823dbf60885c8b43ea36a/sdk/search/Azure.Search.Documents/src/Models/FacetResult.cs#L10) and remove them from [here](https://github.com/Azure/azure-sdk-for-net/blob/b8d45e2478f1404bc22823dbf60885c8b43ea36a/sdk/search/Azure.Search.Documents/src/Generated/Models/FacetResult.cs).

Reply:
> However, shouldn't then there be no change in the one generated file? Or, if we are splitting it up, the new file can also be under Generated\Models\ (something like FacetResult.Dictionary.cs).

Ignore this comment. I see now what you are doing and how the file is being reused for the purpose of implementing `IReadOnlyDictionary`.